storage/url/url.go

```diff
@@ -47,6 +48,7 @@ type URL struct {
 
 // New creates a new URL from given path string.
 func New(s string) (*URL, error) {
+	s = strings.ReplaceAll(s, "\\", "/")
```

Comment:
Why we need this change?

Reply:
in windows os, to allow commands like `cp e2e\app_test.go s3://bucket-faru-1` i.e., backslashed entry

Reply:
This function is used for creating both remote and local URLs. It is not ideal but both S3 and Unix systems may have `\\` in their files/directories. 

For Unix, `t\\est` is a valid pathname. I created an example directory:
```
tree t\\est
t\\est
|-- anotherfilewithbackslashes\\
`-- testfile

0 directories, 2 files
```


For S3, it is also valid:
```
s5cmd ls s3://bigtmp/
                                 DIR k\a/
                                 DIR ka/
2020/07/06 12:22:04                0 newtest
2020/07/06 12:21:40                0 test
```

I suspect if we do backlash replacement on the raw source,  we may affect original user input. For example, for the following command:

```
s5cmd cp test s3://bucket/prefixwith\\backslash/filewith\\backslash
# it will upload into bucket/prefixwith/backslash/filewith/backslash
```

I know it is an edge case (generally we don't use mixed file paths) and won't happen too much,  but still, we need to consider this case :) 


We should do this once we are sure that the OS is WINDOWS and the given URL is not REMOTE(S3 URL).  Maybe [here](https://github.com/peak/s5cmd/blob/master/storage/url/url.go#L53-L62) we may check OS and set URL accordingly. What do you think?

Note: We can add some simple tests for this scenario.

Reply:
Thanks a lot. Thats nice observation and suggestion. will check that too.